src/mysqlsh_helpers.py

```diff
@@ -13,15 +13,38 @@
 logger = logging.getLogger(__name__)
 
 
+class MySQLConfigureMySQLUsersError(Exception):
+    """Exception raised when creating a user fails."""
+
+    pass
+
+
 class MySQL:
     """Class to encapsulate all operations related to the MySQL instance and cluster.
 
     This class handles the configuration of MySQL instances, and also the
     creation and configuration of MySQL InnoDB clusters via Group Replication.
     """
 
-    def __init__(self):
-        pass
+    def __init__(
+        self,
+        root_password: str,
+        cluster_admin_user: str,
+        cluster_admin_password: str,
+        instance_address: str,
+    ):
+        """Initialize the MySQL class.
+
+        Args:
+            root_password: Password for the 'root' user
+            cluster_admin_user: User name for the cluster admin user
+            cluster_admin_password: Password for the cluster admin user
+            instance_address: Address of the targeted instance
+        """
```

Comment:
I think this is going to be irrelevant but bear with me for a second :).

The Juju SDK [best practices about docstrings](https://discourse.charmhub.io/t/charm-development-best-practices/5396#heading--docstrings) says that we follow google docstrings style guide, but [google says](https://google.github.io/styleguide/pyguide.html#384-classes) that class atributtes should be declared under the class, not the `__init__`. Then, I took a look around and I saw many other charms doing it the same way as here (uner `init`). So, no conclusions really, but since I spent time to look for this I wanted to leave the note (:

Reply:
That's a good point of discussion. I would let it on `__init__` just because the IDE's seem to map it on class instantiation when showing inline doc. But I'm not a strong inclined to either, and I appreciate your dedication.